### DISCUSSION TITLE
How to get the position angle of moon's north pole?

### DISCUSSION THREAD

**Discussion by CoolTowel** [skyfielders/python-skyfield#726] [Q&A] [answered]
I am trying to draw a moon phase in the sky, with rough patterns of lunar mares. Is there any approach to get or calculate the position angle moon's north pole, like in this [NASA website](https://svs.gsfc.nasa.gov/4955)?

**Comment by brandon-rhodes** [comment, marked as answer]
That’s a good question! There is probably a more direct way than I'm currently thinking of, but: one approach would be to ask for the position of the Moon in the sky; then ask for the position in the sky of the Moon's north pole as described in [Observing a Moon location](https://rhodesmill.org/skyfield/planetary.html#observing-a-moon-location); and then ask for the position angle between those two sky locations with:

https://rhodesmill.org/skyfield/examples.html#at-what-angle-in-the-sky-is-the-crescent-moon

See if that gives you good numbers that agree with NASA!

**Reply by CoolTowel** [reply]
Thanks for the solution! Really sorry for forgetting to reply. 

I finally used a very rough way to calculate it. I assumed the north pole is fixing to the orbit plane. So the position angle between the position of moon and its position at one hour ago is roughly the position angle of north pole plus 90 degrees. 

Now I am trying to write an iOS app to show the moon phase. I want to use C and Swift to do that. I can make the [SOFA](https://www.iausofa.org/) to work with swift, but it seems there is no function like the Planetary Reference Frames used in [Observing a Moon location](https://rhodesmill.org/skyfield/planetary.html#observing-a-moon-location) in SOFA. I don't want to use the rough way mentioned above. I noticed that the [PyEphem](https://github.com/brandon-rhodes/pyephem) uses a C library. Does it have a similar function like Planetary Reference Frames in skyfield so that I can use it in Swift? (it seems answer is no, but maybe I missed something?)

**Reply by brandon-rhodes** [reply]
Alas, no, the C library that's wrapped by PyEphem doesn't support reference frames.

I would suggest creating a table of dates and phases with Skyfield, and embedding it in the app as a data structure. Moon phases for, say, the next couple of centuries, should be much smaller than the library and data that would be required to compute them. Your app will be smaller, faster, and more reliable, since you will be assured that each new update has the exact same table of Moon phases; you won't be able to accidentally break that calculation, because it's made ahead of time.

**Reply by CoolTowel** [reply]
Thank you for the reply! Calculating it in advance is a good way. BTW, I found this [article](https://astro.ukho.gov.uk/data/tn/naotn74.pdf) introducing how to calculate position angle and other quantities from the JPL lunar ephemerides.
